### PR TITLE
IBX-5033 REST search criteria

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-5033
| Versions      | 4.3+
| Edition       | N/A

image

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/search/criteria_reference/catalogidentifier_criterion.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# CatalogIdentifier Criterion
+
+The `CatalogIdentifier` Search Criterion searches for catalogs by the value of their identifier.
```
Comment (adriendupuis) [inline docs/search/criteria_reference/catalogidentifier_criterion.md:3]: As [there can be only one](https://doc.ibexa.co/en/latest/pim/catalog_api/#get-catalog), maybe singular can remind that an identifier is unique. 

```suggestion
The `CatalogIdentifier` Search Criterion searches for a catalog by the value of its identifier.
```

File: docs/search/criteria_reference/ancestor_criterion.md [hunk 2]
```diff
@@ -9,10 +9,34 @@ searches for content that is an ancestor of the provided Location, including thi
...
+    ```
+
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "AncestorCriterion": "/81/82/"
+            }
+        }
```
Comment (adriendupuis) [inline docs/search/criteria_reference/ancestor_criterion.md:37]: Indent offset

```suggestion
    "Query": {
        "Filter": {
            "AncestorCriterion": "/81/82/"
        }
    }
```

File: docs/search/criteria_reference/contentid_criterion.md [hunk 3]
```diff
@@ -9,6 +9,30 @@ searches for content by its ID.
...
+    ```
+
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "ContentIdCriterion": [69, 72]
+            }
+        }
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contentid_criterion.md:37]: indent offset

```suggestion
    "Query": {
        "Filter": {
            "ContentIdCriterion": [69, 72]
        }
    }
```

File: docs/search/criteria_reference/contenttypegroupid_criterion.md [hunk 4]
```diff
@@ -9,10 +9,34 @@ searches for content based on the ID of its Content Type group.
...
+    ```
+
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "ContentTypeGroupIdCriterion": [1, 2]
+            }
+        }
```
Comment (adriendupuis) [inline docs/search/criteria_reference/contenttypegroupid_criterion.md:37]: indent offset

```suggestion
    "Query": {
        "Filter": {
            "ContentTypeGroupIdCriterion": [1, 2]
        }
    }
```

File: docs/search/criteria_reference/ancestor_criterion.md (hunk 2 above)
Comment (adriendupuis) [reply docs/search/criteria_reference/ancestor_criterion.md]: Fixed in a608648f3e2e44b509be6e68e2a4993cd0aafcce

File: docs/search/criteria_reference/contentid_criterion.md (hunk 3 above)
Comment (adriendupuis) [reply docs/search/criteria_reference/contentid_criterion.md]: Fixed in a608648f3e2e44b509be6e68e2a4993cd0aafcce

File: docs/search/criteria_reference/contenttypegroupid_criterion.md (hunk 4 above)
Comment (adriendupuis) [reply docs/search/criteria_reference/contenttypegroupid_criterion.md]: Fixed in a608648f3e2e44b509be6e68e2a4993cd0aafcce

File: docs/search/criteria_reference/createdat_criterion.md [hunk 5]
```diff
@@ -4,11 +4,13 @@ The `CreatedAt` Search Criterion searches for products based on the date when th
 
 ## Arguments
 
-- `createdAt` - indicating the date that should be matched, provided as a `DateTimeInterface` object
+- `createdAt` (PHP), `created_at` (REST) - indicating the date that should be matched, provided as a `DateTimeInterface` object
```
Comment (adriendupuis) [inline docs/search/criteria_reference/createdat_criterion.md:7]: REST doesn't use an object as PHP does.

[The REST parser get a string and pass it to `DateTime` constructor](https://github.com/ibexa/product-catalog/blob/main/src/bundle/REST/Input/Parser/Query/Criterion/CreatedAt.php#L38). I don't know how to handle this. Maybe something like:

```suggestion
- `createdAt` (PHP), `created_at` (REST) - indicating the date that should be matched, provided as a `DateTimeInterface` object in PHP, or as a string acceptable by `DateTime` constructor in REST
```

File: docs/search/criteria_reference/createdat_criterion.md [hunk 6]
```diff
@@ -4,11 +4,13 @@ The `CreatedAt` Search Criterion searches for products based on the date when th
...
 - `operator` - Operator constant (EQ, GT, GTE, LT, LTE)
```
Comment (adriendupuis) [inline docs/search/criteria_reference/createdat_criterion.md:8]: Same problem of PHP vs REST here. While PHP can use constants, REST must use [the values of those constants](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/Operator.php#L18-L22).

Maybe something like:

```suggestion
- `operator` - Operator constant (EQ, GT, GTE, LT, LTE) in PHP or its value in REST
```

Comment (adriendupuis) [reply docs/search/criteria_reference/createdat_criterion.md:8]: The examples are so naturally talkative in comparison.

**Review by bdunogier — APPROVED** [review]

**Review by adriendupuis — APPROVED** [review]

**Review by julitafalcondusza — APPROVED** [review]

**Pull request merged**